### PR TITLE
build: allow Numpy 1.X

### PR DESCRIPTION
**Pull request by RobPasMue**
Can we not allow NumPy 1.X? This is breaking the rest of the ecosystem @koubaa

### CODE REVIEW

**Comment by RobPasMue** [comment]
https://github.com/ansys/pyansys/pull/764

**Comment by RobPasMue** [comment]
DPF is working on making that upper limit more flexible but setting a minimum of 2.X is pretty constraining

**Comment by RobPasMue** [comment]
This will require a follow up patch release

**Comment by RobPasMue** [comment]
> maybe we can throw an error in the `__init__.py` file of the pydyna-keywords module until hollerith can support both numpy versions?

Works for me. I can add that disclaimer. Is there a big limitation on the numpy version for hollerith?

**Comment by koubaa** [comment]
@RobPasMue see https://github.com/ansys/hollerith/issues/34

**Comment by RobPasMue** [comment]
We might have a bigger issue ... hollerith will force us to install numpy 2.X still. We need to get that one solved first.

**Comment by koubaa** [comment]
> We might have a bigger issue ... hollerith will force us to install numpy 2.X still whenever we pip install... making the warning totally useless. We need to get that one solved first.

Agreed. I may need a bit of help there

File: pyproject.toml [hunk 1]
```diff
@@ -30,7 +30,7 @@ dependencies = ["ansys-dpf-core>=0.7.2",
                 "ansys-platform-instancemanagement~=1.0",
                 "pyvista>=0.43.4",
                 "hollerith==0.4.1",
-                "numpy>=2.0.0",
```
Comment (koubaa) [inline pyproject.toml:33]: @RobPasMue should we pin a low version of numpy based on their support lifecycle? how do we do this in other libraries?

Comment (RobPasMue) [reply pyproject.toml:33]: That would be a good idea, indeed. But currently there's no standard AFAIK. I can have a look though.

File: pyproject.toml [hunk 2]
```diff
@@ -29,15 +29,16 @@ dependencies = ["ansys-dpf-core>=0.7.2",
...
+                "hollerith>=0.4.1",
+                "numpy>=1",
                 "pandas>=2.0",
                 "appdirs>=1.4.4"
 ]
 
 [project.optional-dependencies]
 tests = [
     "matplotlib==3.9.2",
+    "hollerith==0.5.0",
```
Comment (RobPasMue) [inline pyproject.toml:41]:
```suggestion
    "hollerith==0.5.2",

```

**Comment by RobPasMue** [comment]
This will allow to use Numpy 1.X now... releasing right after CI/CD passes

**Pull request merged**